zipline/assets/assets.py

```diff
@@ -106,9 +104,6 @@ def __init__(self, engine):
         for table_name in asset_db_table_names:
             setattr(self, table_name, metadata.tables[table_name])
 
-        # Check the version info of the db for compatibility
-        check_version_info(self.version_info, ASSET_DB_VERSION)
```

Comment:
I'm also adding a test to ensure that the finder is verifying version to avoid someone (me) accidentally removing this check again.